### ISSUE TITLE
Anchor "default" font size should be inherited

### ISSUE THREAD

**Issue by ivan-kleshnin** [mantinedev/mantine#1731]
### What package has an issue

@mantine/core

### Describe the bug

`Anchor` component is hardcoded to be of `md` size by default. 

When it's used within an context of resized text (`h1/Title`, resized `Text`, etc) it continues to have its own (md) size and
each time should be corrected manually. 

Pls. check the [Sandbox](https://codesandbox.io/s/mantine-forked-3xc8fe?file=/src/App.tsx).

### In which browser did the problem occur

All

### Possible fix

`font-weight` behaves correctly. `font-size` should inherit parent size by default in the same way. 

---

Same problem with `Text`. Mantine typography components are not composable 😨 

```
<Text size="lg">
  Large text <Text weight={600}>bold text but small!</Text>
</Text>
```

---

[MUI Sandbox](https://codesandbox.io/s/material-ui-typography-examples-forked-l5768n?file=/index.js). Default behavior is to inherit which preserves composability. Related discussion: https://github.com/mui/material-ui/issues/16110

**Comment by ivan-kleshnin** [comment]
## Updates

1. There's an `inherit` property but it has a higher priority than `size` so it's rather a hard reset.

2. It's possible to override `Text` but it's a lot of code and you'll makes things messy with names and imports.

```tsx
...

export const Text: TextComponent = React.forwardRef(({size, sx, ...rest}: TextProps<"div">, ref: PolymorphicRef<"div">) => {
  sx = size ? sx : {fontSize: "inherit", ...sx} // !!!

  return <M.Text
    {...rest}
    ref={ref}
    size={size}
    sx={sx}
  />
}) as any
```

3. First attempt to override globally (fail)

```
<MantineProvider
  defaultProps={{
    Text: {sx: {fontSize: "inherit"}}, // -- will have more priority than size
  }}
```

4. Second attempt to override globally (success)

```
<MantineProvider
  defaultProps={{
    Text: {size: "whatever"}, // -- any value to "break" the initial "md" setting
  }}
```

5. Before and After

<img width="914" alt="Screen Shot 2022-07-05 at 09 56 43" src="https://user-images.githubusercontent.com/2128182/177280074-42ad8f1b-3c6b-4ac2-882f-972751d2b65a.png">

**Comment by rtivital** [comment]
Fixed in 5.0
